pkg/scanners/clairify/clairify.go

```diff
@@ -451,7 +453,10 @@ func retryOnGRPCErrors(ctx context.Context, name string, f func() error) error {
 	return backoff.RetryNotify(op, backoff.WithContext(eb, ctx), notify)
 }
 
-func (c *clairify) GetNodeInventoryScan(node *storage.Node, inv *storage.NodeInventory) (*storage.NodeScan, error) {
+func (c *clairify) GetNodeInventoryScan(node *storage.Node, inv *storage.NodeInventory, ir *v4.IndexReport) (*storage.NodeScan, error) {
+	if inv == nil && ir != nil {
+		return nil, errors.New("Received a Scanner v4 request for Scanner v2")
```

Comment:
There is a lot of potential to add useful information for the users in this message - we could add that this is combination is incompatible and nudge users to upgrade their secured clusters. If that is impossible, we could suggest a deployment of Central with both scanners simultaneously.

Reply:
At least with the current codebase, it is impossible for this situation to happen, but I agree.
I've updated the error message to nudge upgrading, though.